### PR TITLE
blog post on active learning / uncertainty sampling

### PR DESCRIPTION
**Pull request by ted-positronix**
A blog post about using active learning and uncertainty sampling to reduce customers' labeling costs.

### CODE REVIEW

**Review by tyler-romero — APPROVED** [review]
LGTM, nice work ted

File: docs/blog/2024-05-01-active-learning.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
+---
+title: "Reducing Labeling Costs with Active Learning"
+description: How Groundlight uses active learning to train accurate vision models while saving on data labeling costs.
+slug: active-learning
+authors:
+  - name: Ted Sandler
+    title: Senior Applied Scientist
+    image_url: https://a-us.storyblok.com/f/1015187/1000x1000/efc35da152/sandlert.jpg
+tags: [active learning, uncertainty sampling, deep dive]
+image: ./images/active-learning/social.png
```
Comment (robotrapta) [inline docs/blog/2024-05-01-active-learning.md:10]: I'm told that normal humans don't like images that are obviously generated by AI for social posts.  My bad for suggesting such a travesty.

File: docs/blog/2024-05-01-active-learning.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+The training sets of all three models are initialized with the same 20 images, 10 labeled in each class.
+
+The plot below shows that the model trained with moderate uncertainty sampling (confidence threshold 95%) has an error rate similar to the model trained without any uncertainty sampling (confidence threshold 100%). This demonstrates that uncertainty sampling can be just as accurate as training models on all available data.
+<img
+ src={require('./images/active-learning/error-rate-over-time.png').default} 
+ width="500 px"
+/>
+
+On the other hand, aggressive uncertainty sampling (confidence threshold 75%) does not escalate enough images for labeling, resulting in a model trained on less data that makes more mistakes. This shows how the confidence threshold controls the trade off between model accuracy and human labeling cost. Indirectly, it also demonstrates the importance of calibrating models so that their reported probabilities reflect observed frequencies and can be used as decision criteria. We calibrate machine learning models at Groundlight, though the details of model calibration are beyond the scope of this post.
```
Comment (robotrapta) [inline docs/blog/2024-05-01-active-learning.md:58]: I like this wording.

File: docs/blog/2024-05-01-active-learning.md [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+As an example, the images shown below were sent to a detector that identifies the presence of dogs in and around an indoor swimming pool. The model reports with 95% confidence that there is a dog in the image on the left. But it is less confident in its response for the image on the right, saying there is no dog present with only 75% confidence. There is in fact a dog in that image, at the back left corner of the swimming pool, but it is difficult to see.
```
Comment (robotrapta) [inline docs/blog/2024-05-01-active-learning.md:24]: Should we hyperlink to our camera feed provider?  Not necessary I think, but maybe polite.

**Review by robotrapta — APPROVED** [review]
Generally looks great to me.  A couple tweaks on the edges.

File: docs/blog/2024-05-01-active-learning.md (hunk 3 above)
Comment (ted-positronix) [reply docs/blog/2024-05-01-active-learning.md:24]: yes, good idea. will do

File: docs/blog/2024-05-01-active-learning.md [hunk 4]
```diff
@@ -0,0 +1,75 @@
+---
+title: "Reducing Labeling Costs with Active Learning"
+description: filler
```
[the remark this answers is not in the document]
Comment (ted-positronix) [reply docs/blog/2024-05-01-active-learning.md]: done

**Pull request merged**